src/DiagnosticsManager.ts

```diff
@@ -92,108 +100,161 @@ export class DiagnosticsManager implements vscode.Disposable {
      * @param uri {@link vscode.Uri Uri} of the file these diagonstics apply to
      * @param sources The source of the diagnostics which will apply for cleaning
      * up diagnostics that have been removed. See {@link swiftc} and {@link sourcekit}
-     * @param diagnostics Array of {@link vscode.Diagnostic}. This can be empty to remove old diagnostics for the specified `sources`.
+     * @param newDiagnostics Array of {@link vscode.Diagnostic}. This can be empty to remove old diagnostics for the specified `sources`.
      */
-    handleDiagnostics(uri: vscode.Uri, sources: string[], diagnostics: vscode.Diagnostic[]): void {
+    handleDiagnostics(
+        uri: vscode.Uri,
+        sources: string[],
+        newDiagnostics: vscode.Diagnostic[]
+    ): void {
+        const isFromSourceKit = !!DiagnosticsManager.sourcekit.find(s => sources.includes(s));
         // Is a descrepency between SourceKit-LSP and older versions
         // of Swift as to whether the first letter is capitalized or not,
         // so we'll always display messages capitalized to user and this
         // also will allow comparing messages when merging
-        diagnostics.forEach(this.capitalizeMessage);
-        const newDiagnostics = this.diagnosticCollection.get(uri)?.slice() || [];
+        newDiagnostics.forEach(this.capitalizeMessage);
```

Comment:
nit: Swift has taught me to prefer the functional style where possible: `newDiagnostics = newDiagnostics.map(this.capitalizeMessage)` where `capitalizeMessage` returns a copy of the object with the message field updated.

Reply:
done